File: EasyStock.Client/easystock-client/src/app/shared/column-meta-data.ts
export interface ValidationRules {
  required?: boolean;
  maxLength: number;
  minLength: number;
  min: number;
  max: number;
  isPassword: boolean;
  isEmail: boolean;
  isUrl: boolean;
  pattern: string;
}

export interface ColumnMetaData {
  name: string;
  type: string;
  isEditable: boolean;
  isFilterable: boolean;
  isSortable: boolean;
  displayName?: string;
  isLookup: boolean;
  lookupIdField?: string;
  isOnlyDetail: boolean;
  validationRules?: ValidationRules;
}

File: EasyStock.API/EasyStock.API/Common/ColumnMetaData.cs
namespace EasyStock.API.Common
{
    public class ColumnMetaData
    {
        public string Name { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty; 
        public bool IsEditable { get; set; }
        public bool IsFilterable { get; set; }
        public bool IsSortable { get; set; }
        public string? DisplayName { get; set; }
        public bool IsLookup { get; set; } = false;
        public string? LookupIdField { get; set; }
        public ValidationRules ValidationRules { get; set; } = new ValidationRules();
    }
}
